### ISSUE TITLE
[spv] spv-out outputs invalid SPIR-V code, and spv-in accepts it

### ISSUE THREAD

**Issue by hanawatson** [gfx-rs/naga#1948]
Translating the following piece of WGSL code to SPIR-V with naga spv-out returns no errors, and neither does validating the translated shader with naga spv-in. However, putting the translated shader through [spirv-val (SPIRV-Tools)](https://github.com/KhronosGroup/SPIRV-Tools) throws an error. I believe it is also invalid SPIR-V according to the specification.

WGSL (also applies for any other Boolean expression other than `true`):
```
@stage(compute) @workgroup_size(1)
fn main() {
	var var0 = bool(true);
}
```

Error from spirv-val:
```
        error: line 23: Expected Result Type to be a pointer or int or float vector or scalar type: Bitcast
          %15 = OpBitcast %bool %14
```

**Comment by JCapucho** [comment]
`spv-in` accepting bad spirv is normal, since naga doesn't make any promise of validating spirv rules (wgsl rules are still validated), the spv-out issue is real though

**Comment by JCapucho** [comment]
Fixed in 7512d4cb69a41397945642c543c634d068e56df2
